android/docs/engine/KMManager/setBannerOptions.md

```diff
@@ -0,0 +1,52 @@
+---
+title: KMManager.setBannerOptions()
+---
+
+## Summary
+
+The **`setBannerOoptions()`** method sends options to the KeymanWeb engine for controlling whether to generate predictions.
+
+## Syntax
+
+``` javascript
+KMManager.setBannerOptions(boolean mayPredict, KeyboardType keyboardType)
+```
+
+### Parameters
+
+`mayPredict`
+:   If `false`, KeymanWeb engine will disable predications and not show suggestions on the banner.
+
+`keyboardType`
+:   The keyboard type. `KEYBOARD_TYPE_INAPP` or
+    `KEYBOARD_TYPE_SYSTEM`.
+
+## Description
+
+Use this method to enable or disable the KeymanWeb engine from generating and displaying predictions on the suggestion banner.
+
+## Examples
+
+### Example: Using `setBannerOptions()`
+
+The following script illustrate the use of `setBannerOptions()`:
+
+``` javascript
+
+    // Temporarily disable predictions on certain fields (e.g. hidden password field or numeric)
+    inputType = attribute.inputType;
+    KMManager.setPredictionsSuspended(inputType, KeyboardType.KEYBOARD_TYPE_SYSTEM);
+    if (KMManager.getPredictionsSuspended(KeyboardType.KEYBOARD_TYPE_SYSTEM)) {
+      KMManager.setBannerOptions(false, KeyboardType.KEYBOARD_TYPE_SYSTEM);
+      // Set the system keyboard HTML banner
+      BannerController.setHTMLBanner(this, KeyboardType.KEYBOARD_TYPE_SYSTEM);
+    }
```

Comment:
This seems unwieldy. Why do we need to tell `KMManager` to disable predictions via `setBannerOptions` and also via `setPredictionsSuspended`? Why doesn't `setPredictionsSuspended` update the banner itself? (And what is the difference between `setPredictionsSuspended` and `setBannerOptions` in any case?)

Reply:
setPredictionsSuspended - sets the internal KMManager vars `inAppPredictionsSuspendedForSensitiveInput` and `systemPredictionsSuspendedForSensitiveInput`

setBannerOptions - passes `mayPredict` to KeymanWeb which controls generating suggestions

We could fold setBannerOptions into setPredictionsSuspended, but the Keyman Engine doesn't have the SystemKeyboard visibility to set the HTML banner (update KeymanWeb with the banner assets)